x/ibc/02-client/types/tendermint/evidence.go

```diff
@@ -51,41 +51,53 @@ func (ev Evidence) Hash() cmn.HexBytes {
 
 // ValidateBasic implements Evidence interface
 func (ev Evidence) ValidateBasic() error {
```

Comment:
I believe this does all checks here:

```typescript

// misbehaviour verification function defined by the client type
// any duplicate signature by a past or current key freezes the client
function checkMisbehaviourAndUpdateState(
  clientState: ClientState,
  evidence: Evidence) {
    h1 = evidence.h1
    h2 = evidence.h2
    abortTransactionUnless(clientState.pastPublicKeys.contains(h1.publicKey))
    abortTransactionUnless(h1.sequence === h2.sequence)
    abortTransactionUnless(h1.commitmentRoot !== h2.commitmentRoot || h1.publicKey !== h2.publicKey)
    abortTransactionUnless(h1.publicKey.verify(h1.signature))
    abortTransactionUnless(h2.publicKey.verify(h2.signature))
    clientState.frozen = true
}
```

except for this check:
```typescript
    abortTransactionUnless(clientState.pastPublicKeys.contains(h1.publicKey))
```

Need reviewers to double-check that this claim is true

Reply:
Hmm - this was an example in the spec, not the actual check for Tendermint.

We want to freeze the client iff. there are two valid headers at the same height.

@melekes Is this logic written out somewhere?

Reply:
Not yet. There's a fork accountability spec https://github.com/tendermint/tendermint/blob/master/docs/spec/consensus/fork-accountability.md.

> We want to freeze the client iff. there are two valid headers at the same height.

yeah. so basically we stop the client if we discovered (either by cross-checking the headers OR receiving evidence) that the light client is under attack. But even then, we can probably try to continue with an alternative provider.